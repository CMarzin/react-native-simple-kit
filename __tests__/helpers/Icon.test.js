import '@mocks/SVG'
import 'react-native'
import renderer from 'react-test-renderer'
import getIcon from '@helpers/icon'

describe('Helper function: getIcon', () => {
  test('renders correctly', () => {
    const component = renderer.create(getIcon('backButton', 100))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('renders with actived parameter', () => {
    const component = renderer.create(getIcon('backButton', 100, true))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('renders with color parameter', () => {
    const component = renderer.create(
      getIcon('backButton', 100, false, '#122E46'),
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
