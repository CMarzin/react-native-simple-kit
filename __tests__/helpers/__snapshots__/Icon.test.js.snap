// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper function: getIcon renders correctly 1`] = `
<View
  actived={false}
  defaultColor="#b8b8b8"
  defaultactivedColor={undefined}
  name="backButton"
  size={100}
  style={null}
>
  <Svg />
</View>
`;

exports[`Helper function: getIcon renders with actived parameter 1`] = `
<View
  actived={true}
  defaultColor="#b8b8b8"
  defaultactivedColor={undefined}
  name="backButton"
  size={100}
  style={null}
>
  <Svg />
</View>
`;

exports[`Helper function: getIcon renders with color parameter 1`] = `
<View
  actived={false}
  defaultColor="#122E46"
  defaultactivedColor={undefined}
  name="backButton"
  size={100}
  style={null}
>
  <Svg />
</View>
`;
